src/main/java/dev/emortal/velocity/player/UsernameSuggestions.java

```diff
@@ -0,0 +1,65 @@
+package dev.emortal.velocity.player;
+
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.suggestion.SuggestionProvider;
+import com.mojang.brigadier.suggestion.Suggestions;
+import com.mojang.brigadier.suggestion.SuggestionsBuilder;
+import com.velocitypowered.api.command.CommandSource;
+import com.velocitypowered.api.proxy.Player;
+import dev.emortal.api.grpc.mcplayer.McPlayerProto.SearchPlayersByUsernameRequest.FilterMethod;
+import dev.emortal.api.model.common.Pageable;
+import dev.emortal.api.model.mcplayer.McPlayer;
+import dev.emortal.api.service.mcplayer.McPlayerService;
+import io.grpc.StatusRuntimeException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+public final class UsernameSuggestions {
+    private static final Logger LOGGER = LoggerFactory.getLogger(UsernameSuggestions.class);
+
+    private final McPlayerService playerService;
+
+    public UsernameSuggestions(@Nullable McPlayerService playerService) {
+        this.playerService = playerService;
+    }
+
+    public @NotNull SuggestionProvider<CommandSource> command(@NotNull FilterMethod filterMethod) {
+        return (context, builder) -> this.command(context, builder, filterMethod);
+    }
+
+    private @NotNull CompletableFuture<Suggestions> command(@NotNull CommandContext<CommandSource> context, @NotNull SuggestionsBuilder builder,
+                                                           @NotNull FilterMethod filterMethod) {
+        if (!(context.getSource() instanceof Player player)) return CompletableFuture.completedFuture(builder.build());
+        if (this.playerService == null) return CompletableFuture.completedFuture(builder.build());
+
+        String currentInput = builder.getRemainingLowerCase();
+        if (currentInput.length() < 3) return CompletableFuture.completedFuture(builder.build());
```

Comment:
NGL I think we change the min input to 2 before suggestions. It'll feel a bit more reactive.

I'd do it myself if I could rn